backend/lucene/src/main/java/org/hibernate/search/backend/lucene/impl/LuceneBackendFactory.java

```diff
@@ -165,6 +175,18 @@ private MultiTenancyStrategy getMultiTenancyStrategy(ConfigurationPropertySource
 		}
 	}
 
+	@SuppressWarnings("deprecation")
+	private LuceneIdReaderWriter getIdReaderWriter(ConfigurationPropertySource propertySource) {
+		LuceneIdReaderWriter.Strategy strategy = SCHEMA_ID_STRATEGY.get( propertySource );
+
+		if ( LuceneIdReaderWriter.Strategy.LUCENE_8.equals( strategy ) ) {
+			log.usingDeprecatedSchemaIdStrategy( SCHEMA_ID_STRATEGY.resolveOrRaw( propertySource ),
+					LuceneIdReaderWriter.Strategy.LUCENE_9.externalRepresentation() );
+		}
```

Comment:
We can't exactly print a warning when using `lucene8` because:

1. It's the default.
2. At the moment it's perfectly reasonable to keep using this strategy. I expect only WildFly will ever make use of `lucene9` TBH.

Reply:
mmm OK 😃. I'll drop the warning.  I suppose there's no SUGGESTION logger level we could use 😄. Yeah ... I was just hoping that if we log it, someone might notice it and will decide to prepare for the upgrade before actually upgrading to 7 😃 And  I thought we could get away with a warning since that is an old behaviour we are about to change...